utils/perl5lib/Batch/BatchMaker.pm

```diff
@@ -549,7 +571,42 @@ sub setCESMRun()
 	if(! defined $chosenmpielem)
 	{
 		my @defaultmpielems = $configmachinesparser->findnodes("/config_machines/machine[\@MACH=\'$self->{'machine'}\']/mpirun[\@mpilib=\'default\']");
-		$chosenmpielem = $defaultmpielems[0];
+		foreach my $defelem(@defaultmpielems)
+		{
+			if(! $defelem->hasAttributes() )
+			{
+				$chosenmpielem = $defelem;
+			}	
+			else
+			{
+				my $attrMatch = 1;
+				my @attrs = $defelem->getAttributes();
+				foreach my $attr(@attrs)
+				{
+					my $attrName = $attr->getName();
+					my $attrValue = $attr->getValue();
+					next if($attrValue eq 'default');
+					my $lcAttrName = lc $attrName;
+					if(defined $self->{$lcAttrName} && (lc $self->{$attrName} eq lc $attrValue))
+					{
+						$attrMatch = 1;
+						next;
+					}
```

Comment:
Can't we remove the block above? If we enter that block, `$attrMatch` will always be `1`, so we shouldn't need to change it. Also, if the conditional above succeeds, then the one below will fail and we will just move on to the next iteration, so we shouldn't need a `next`, unless I'm missing something.